web/static/js/components/thesis-editor.js

```diff
@@ -312,6 +317,38 @@ class ThesisEditor extends React.Component {
     this.setState({importProgress: progress})
   }
 
+  checkForDuplicateContentBlocks () {
+    const editors = this.allContentEditors()
+
+    const names = {}
+    for (let i = 0; i < editors.length; i++) {
+      const name = editors[i].dataset.thesisContentId
+      if (names[name]) {
+        names[name] += 1
+      } else {
+        names[name] = 1
+      }
+    }
+
+    const duplicateIndex = Object.values(names).findIndex((value) => {
+      return value > 1
+    })
+
+    if (duplicateIndex > 0) {
+      const key = Object.keys(names)[duplicateIndex]
+
+      this.setState({errorAlertText: `
+        Content block: '${key}' occurs on the page more than once.
+        This will cause issues and must be resolved. Check that each content
+        block in your template has a unique name.
+      `.replace(/^\s+|\s+$|\s+(?=\s)/g, '')})
+    }
```

Comment:
does this set state with a diff name every loop? it may not be a good user experience if there are like 15 blocks with the same name and 15 alerts one right after the other one is dismissed.